### PR TITLE
Stop hawk service when removing node

### PR DESCRIPTION
**Pull request by liangxin1300**
## Problem
crmsh should stop hawk.service when removing node

### CODE REVIEW

File: crmsh/bootstrap.py [hunk 1]
```diff
@@ -2152,15 +2152,24 @@ def set_cluster_node_ip():
             break
 
 
+def stop_cluster_related_service(remote_addr=None):
```
Comment (MalloZup) [inline crmsh/bootstrap.py:2155]: def stop_cluster_related_service(services, remote_addr=None):
    """
    Disable and stop cluster related service
    """
    for service in services:
        if utils.service_is_available(service, remote_addr=remote_addr):
            status("Stopping the {}".format(service))
            utils.stop_service(service, disable=True, remote_addr=remote_addr)

File: crmsh/bootstrap.py [hunk 2]
```diff
@@ -2424,7 +2433,7 @@ def remove_self():
             error("Failed to remove this node from {}".format(othernode))
     else:
         # disable and stop cluster
-        utils.stop_service("corosync", disable=True)
+        stop_cluster_related_service()
```
Comment (MalloZup) [inline crmsh/bootstrap.py:2436]: services = ["corosync-qdevice.service", "corosync.service", "hawk.service"]:
stop_cluster_related_service(services)

**Review by MalloZup — CHANGES_REQUESTED** [review]
looks good
I would separate the Input from a function to make it more stateless and generic.


So in the example the function should not known about the services but they should be an arg

**Comment by liangxin1300** [comment]
> looks good
> I would separate the Input from a function to make it more stateless and generic.
> 
> So in the example the function should not known about the services but they should be an arg

Thanks @MalloZup ! Already changed!

File: crmsh/bootstrap.py [hunk 3]
```diff
@@ -2152,15 +2152,24 @@ def set_cluster_node_ip():
             break
 
 
+def stop_cluster_related_service(stop_list, remote_addr=None):
```
Comment (arbulu89) [inline crmsh/bootstrap.py:2155]: As this works now as a generic method, maybe it should be renamed to `stop_services` and receive `service` as argument.

File: crmsh/bootstrap.py [hunk 4]
```diff
@@ -44,7 +44,7 @@
 RSA_PRIVATE_KEY = "/root/.ssh/id_rsa"
 RSA_PUBLIC_KEY = "/root/.ssh/id_rsa.pub"
 AUTHORIZED_KEYS_FILE = "/root/.ssh/authorized_keys"
-
+SERVICES_STOP_LIST = ["corosync-qdevice.service", "corosync.service", "hawk.service"]
```
Comment (arbulu89) [inline crmsh/bootstrap.py:47]: I hope this won't bring some issue during the backports as `corosync-qdevice.service` doesn't exist in some branches

**Review by arbulu89 — APPROVED** [review]
@liangxin1300 It looks good. I proposed some renaming of methods, but up to you if you like them or not.

File: crmsh/bootstrap.py (hunk 3 above)
Comment (MalloZup) [reply crmsh/bootstrap.py:2155]: yep I agree with naming

File: crmsh/bootstrap.py (hunk 4 above)
Comment (liangxin1300) [reply crmsh/bootstrap.py:47]: I will notice to remove `corosync-qdevice.service` in other branches

Comment (MalloZup) [reply crmsh/bootstrap.py:47]: we should then check if the service exists on the function name and don't errror. good catch

File: crmsh/bootstrap.py (hunk 3 above)
Comment (liangxin1300) [reply crmsh/bootstrap.py]: OK, already change to `stop_services`

File: crmsh/bootstrap.py (hunk 4 above)
Comment (arbulu89) [reply crmsh/bootstrap.py:47]: Maybe it's not even needed. The method already checks if the service is available. What happens if it's not? It fails? Or it just continues? If it continues, we would be fine

Comment (MalloZup) [reply crmsh/bootstrap.py:47]: yep agree if it continues is no problem . is just we shouldn't raise exceptions

**Pull request merged**